File: users-service/src/routes/health.js
const express = require('express');
const router = express.Router();
const { pool } = require('../config/database');

// Health check endpoint
router.get('/', (req, res) => {
  res.json({
    status: 'healthy',
    service: 'users-service',
    timestamp: new Date().toISOString(),
    version: process.env.npm_package_version || '1.0.0'
  });
});

// Readiness check
router.get('/ready', async (req, res) => {
  try {
    // Check database connection
    await pool.query('SELECT 1');
    res.json({
      status: 'ready',
      service: 'users-service',
      database: 'connected',
      timestamp: new Date().toISOString()
    });
  } catch (error) {
    res.status(503).json({
      status: 'not ready',
      service: 'users-service',
      database: 'disconnected',
      error: error.message,
      timestamp: new Date().toISOString()
    });
  }
});

// Liveness check
router.get('/live', (req, res) => {
  res.json({
    status: 'alive',
    service: 'users-service',
    uptime: process.uptime(),
    timestamp: new Date().toISOString()
  });
});

module.exports = router;

File: products-service/src/routes/health.py
from flask import Blueprint, jsonify
import os
from datetime import datetime

from config.database import get_database

health_bp = Blueprint('health', __name__)

@health_bp.route('/', methods=['GET'])
def health_check():
    return jsonify({
        'status': 'healthy',
        'service': 'products-service',
        'timestamp': datetime.utcnow().isoformat(),
        'version': '1.0.0',
        'message': 'Products service is running perfectly! 🎯'
    })

@health_bp.route('/ready', methods=['GET'])
def readiness_check():
    try:
        # Check database connection
        db = get_database()
        db.command('ping')
        
        return jsonify({
            'status': 'ready',
            'service': 'products-service',
            'database': 'connected',
            'timestamp': datetime.utcnow().isoformat()
        })
    except Exception as e:
        return jsonify({
            'status': 'not ready',
            'service': 'products-service',
            'database': 'disconnected',
            'error': str(e),
            'timestamp': datetime.utcnow().isoformat()
        }), 503

@health_bp.route('/live', methods=['GET'])
def liveness_check():
    return jsonify({
        'status': 'alive',
        'service': 'products-service',
        'timestamp': datetime.utcnow().isoformat()
    })
